test/geometry/epipolar/test_projection.py

```diff
@@ -138,3 +138,54 @@ def test_gradcheck(self, device):
         F_mat = torch.rand(1, 3, 3, device=device, dtype=torch.float64, requires_grad=True)
         assert gradcheck(epi.projections_from_fundamental,
                          (F_mat,), raise_exception=True)
+
+
+class TestKRtFromProjection:
+    def test_smoke(self, device, dtype):
+        P = torch.randn(1, 3, 4, device=device, dtype=dtype)
+        K, R, t = epi.KRt_from_projection(P)
+        assert K.shape() == (1, 3, 3)
+        assert R.shape() == (1, 3, 3)
+        assert t.shape() == (1, 3, 1)
+
+    @pytest.mark.parametrize("batch_size", [1, 2, 4])
+    def test_shape(self, batch_size, device, dtype):
+        B: int = batch_size
+        P = torch.rand(B, 3, 3, device=device, dtype=dtype)
+        K, R, t = epi.KRt_from_projection(P)
+        assert K.shape() == (B, 3, 3)
+        assert R.shape() == (B, 3, 3)
+        assert t.shape() == (B, 3, 1)
+
+    def test_simple(self, device, dtype):
+        P = torch.Tensor([[
+            [232., 71., 267., 352.]
+            [280., 150., 140., 198.]
+            [158., 468., 237., 371.]
+        ]], device=device, dtype=dtype)
+
+        K_expected = torch.Tensor([[
+            [154.8365, 217.0313, 243.0576],
+            [0.0, 218.8586, 269.4455],
+            [0.0, 0.0, 547.8659]
+        ]], device=device, dtype=dtype)
+
+        R_expected = torch.Tensor([[
+            [-0.2499, -0.3690, 0.8952],
+            [0.9243, -0.3663, 0.1071],
+            [0.2884, 0.8542, 0.4326]
+        ]], device=device, dtype=dtype)
+
+        t_expected = torch.tensor([
+            [[0.0167], [-0.1426], [-1.2950]],
+        ], device=device, dtype=dtype)
+
+        K_estimated, R_estimated, t_estimated = epi.KRt_from_projection(P)
+        assert_allclose(K_estimated, K_expected)
+        assert_allclose(R_estimated, R_expected)
+        assert_allclose(t_estimated, t_expected)
```

Comment:
we could add an extra test to check for going forward and backward `KRt_from_projection <-> projection_from_KRt`

Reply:
add the test I mention here so that we can evaluate both directions

Reply:
Hello @edgarriba , since I'm going with replication  of result wrt [this implementation](https://github.com/opencv/opencv_contrib/blob/ab2e83248fa1783d7c2055ef280d403ffcf968c8/modules/sfm/src/libmv_light/libmv/multiview/projection.cc#L32),  the `C-code` actually normalizes the `K-matrix` in the end (to make the last element to be 1). 
And if this `K` is fed to `projection_from_KRt()` as a sanity-check, we will get different result for `P`

So, I will have to leave the `K-matrix` un-normalized to cope with getting the exact `P` out of `projection_from_KRt()`